front/src/applications/operationalStudies/components/ImportTrainSchedule/ImportTrainScheduleModal.jsx

```diff
@@ -206,15 +209,24 @@ export default function ImportTrainScheduleModal(props) {
   async function launchTrainSchedules(params) {
     try {
       await postTrainSchedule({ standaloneSimulationParameters: params }).unwrap();
-      getTimetable(timetableID);
-      return `${t(
-        'operationalStudies/importTrainSchedule:status.calculatingTrainScheduleComplete'
-      )} (${params.path})`;
     } catch (error) {
       return `${t(
         'operationalStudies/importTrainSchedule:errorMessages.unableToRetrieveTrainSchedule'
       )} (${params.path})`;
     }
+
+    try {
+      const timetable = await getTimetableWithTrainSchedulesDetails({
+        id: timetableID,
+      }).unwrap();
+      await Promise.resolve(dispatch(updateReloadTimetable(false)));
```

Comment:
Not sure if this is really safe to rely on a promise. 
this probably works but that’s not how dispatch is intended to be used.

Reply:
This part comes from my PR.
I noticed that during the call of the endpoint, the dispatch wasn't catched soon enough to rerender the trains list and ended up doing this solution.

Maybe in the next PI, I hope to avoid using this kind of dispatch and will try to use rtk queries tags invalidation to rerender the list.